### DISCUSSION TITLE
Auto discovery did not detect a Docker host configuration

### DISCUSSION THREAD

**Discussion by Zombach** [testcontainers/testcontainers-dotnet#1071] [Q&A] [answered]
I'm trying to run tests from CI/CD.
Each test suite creates containers within itself through testcontainers. But I get the error Auto discovery did not detect a Docker host configuration. And System.ArgumentException : Docker is either not running or misconfigured. Please ensure that Docker is running and that the endpoint is properly configured. You can customize your configuration using either the environment variables or the ~/.testcontainers.properties file.
I just can’t figure out what needs to be written down and where. so that it works for me. It is advisable to configure the container.
In fact, I need to connect to the docker api from docker.
Help plis.

I would like to specify the setting either in testcontainers or in yml

testcontainers
<img width="425" alt="image" src="https://github.com/testcontainers/testcontainers-dotnet/assets/52016832/56976e35-4a7f-409a-b83c-9c651b50f434">

yml
<img width="421" alt="image" src="https://github.com/testcontainers/testcontainers-dotnet/assets/52016832/c3bb750b-95f8-48fa-a7ba-8ecf76d1c3b9">

Errors
<img width="590" alt="image" src="https://github.com/testcontainers/testcontainers-dotnet/assets/52016832/44a39269-c341-45da-b622-e1ef32774907">

**Comment by HofmeisterAn** [comment]
The error message indicates that Testcontainers cannot detect a container runtime (Docker) because you are running inside another container. The `mcr.microsoft.com/dotnet/sdk` image does not contain Docker nor does it have access to the host's Docker daemon socket. There are several ways to configure this, but the method depends on your CI/CD provider. For example, with [Azure DevOps](https://dev.azure.com/HofmeisterAn/GitHub/_git/testcontainers-dotnet-679?path=/azure-pipelines.yml), no additional configurations are necessary. It utilizes the [Docker Wormhole](https://dotnet.testcontainers.org/examples/dind/) configuration and automatically mounts the socket. Which CI/CD provider are you using?

**Reply by Zombach** [reply]
I use GitLab and runner git lab

**Reply by HofmeisterAn** [reply, marked as answer]
GitLab requires additional configuration, perhaps [this](https://dotnet.testcontainers.org/cicd/#gitlab-cicd) or [this](https://www.atomicjar.com/2023/01/running-testcontainers-tests-on-gitlab-ci/) article (OC without the Java stuff) helps. There is also a GitLab doc about integration Docker.
